File: models/produto.py
from utils.helper import formata_float_moeda


class Produto:
    contador = 1


    def __init__(self, nome, preco):
        """
        Metodo construtor da classe Produto
        """
        self.__codigo = Produto.contador
        self.__nome = nome
        self.__preco = preco
        Produto.contador += 1

    """
    Getters e Setters de todos os atributos
    """
    @property
    def codigo(self):
        return self.__codigo

    @property
    def nome(self):
        return self.__nome

    @property
    def preco(self):
        return self.__preco

    def __str__(self):
        """
        Sobrescrita do metodo "To String" de um objeto do tipo Produto
        """
        return f'Codigo: {self.codigo}\nNome: {self.nome}\nPreço: {formata_float_moeda(self.preco)}'

File: src/model/Produto.java
package model;

import helper.Utils;

/**
 * Classe Produto e seus atributos
 */
public class Produto {
	private static int contador = 1;
	private int id;
	private String nome;
	private Double preco;
	
	
	
	/**
	 * Metodo construtor da classe Produto
	 */
	public Produto(String nome, Double preco) {
		this.id = contador;
		this.nome = nome;
		this.preco = preco;
		contador ++;
	}
	
	/*
	 * Getters e Setters dos atributos
	 */
	public int getId() {
		return id;
	}

	public String getNome() {
		return nome;
	}
	public void setNome(String nome) {
		this.nome = nome;
	}
	public Double getPreco() {
		return preco;
	}
	public void setPreco(Double preco) {
		this.preco = preco;
	}
	
	/**
	 * Sobrescrita do metodo "To String" de um objeto do tipo Produto
	 */
	public String toString() {
		return "ID: " + this.getId() + "\nNome: " + this.getNome() + "\nPreço: " + Utils.double_string(this.getPreco());
	}
	

}
